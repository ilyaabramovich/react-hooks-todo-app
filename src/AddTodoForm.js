import React, { useState } from 'react'

export const AddTodoForm = ({ onSubmit }) => {
    const [text, setText] = useState("");

    const handleChange = event => {
        setText(event.target.value);
    };

    const handleSubmit = event => {
        event.preventDefault();
        onSubmit(text);
        setText("");
    };

    return (
        <form>
            <div className="form-group">
                <input
                    autoFocus={true}
                    className="form-control"
                    type="text"
                    value={text}
                    onChange={handleChange}
                    placeholder="Enter new TODO"
                />
            </div>
            <button
                type="submit"
                disabled={!text}
                className="submit-button"
                onClick={handleSubmit}
            >
                Add
        </button>
        </form>
    )
}
