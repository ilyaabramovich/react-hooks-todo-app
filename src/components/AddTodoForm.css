.add-todo-form__control {
  margin: 0 auto 1.25rem auto;
}

.add-todo-form__input_fullwidth {
  width: 100%;
}

.add-todo-form__input {
  display: block;
  height: 2.375rem;
  padding: 0.375rem 0.75rem;
  color: var(--color-gray);
  background-color: white;
  background-clip: padding-box;
  border: 1px solid var(--color-lightgray);
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.input {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.add-todo-form__input:focus {
  border-color: #80bdff;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
}

.add-todo-form__button_fullwidth {
  width: 100%;
}

.add-todo-form__button {
  border-radius: 4px;
  display: block;
  background: var(--color-green);
  color: #fff;
  padding: 0.75rem;
}
