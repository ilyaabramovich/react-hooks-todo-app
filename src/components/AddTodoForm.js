import React, { useState } from "react";
import "./AddTodoForm.css";

export const AddTodoForm = ({ onSubmit }) => {
  const [text, setText] = useState("");

  const handleChange = event => {
    setText(event.target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    onSubmit(text);
    setText("");
  };

  return (
    <form className="add-todo-form">
      <div className="add-todo-form__control">
        <input
          autoFocus={true}
          className="input add-todo-form__input add-todo-form__input_fullwidth"
          type="text"
          value={text}
          onChange={handleChange}
          placeholder="Enter new TODO"
        />
      </div>
      <button
        type="submit"
        disabled={!text}
        className="button add-todo-form__button add-todo-form__button_fullwidth"
        onClick={handleSubmit}
      >
        Add
      </button>
    </form>
  );
};
