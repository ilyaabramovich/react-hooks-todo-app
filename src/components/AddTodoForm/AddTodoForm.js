import React, { useState } from "react";
import "./AddTodoForm.css";

export const AddTodoForm = ({ onSubmit }) => {
  const [text, setText] = useState("");

  const handleChange = event => {
    setText(event.target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    onSubmit(text);
    setText("");
  };

  return (
    <form className="add-todo-form" data-testid="form" onSubmit={handleSubmit}>
      <div className="add-todo-form__control">
        <input
          autoFocus
          aria-label="New thing to do"
          className="input add-todo-form__input add-todo-form__input_fullwidth"
          type="text"
          value={text}
          onChange={handleChange}
          placeholder="New thing to do"
        />
      </div>
      <button
        type="submit"
        disabled={!text}
        className="button add-todo-form__button add-todo-form__button_fullwidth"
      >
        Add
      </button>
    </form>
  );
};
