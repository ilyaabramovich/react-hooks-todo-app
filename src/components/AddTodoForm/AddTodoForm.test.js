import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { AddTodoForm } from "./AddTodoForm";

test("it renders", () => {
  const { asFragment } = render(<AddTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

test("renders form with button disabled by default", () => {
  const { getByRole } = render(<AddTodoForm />);
  const button = getByRole("button");
  expect(button).toBeDisabled();
});

test("changing input enables submit button", () => {
  const { getByRole } = render(<AddTodoForm />);
  const input = getByRole("textbox");
  const button = getByRole("button");

  fireEvent.change(input, { target: { value: "a" } });
  expect(button).not.toBeDisabled();
});

it("submits", () => {
  const onSubmit = jest.fn();
  const { getByTestId } = render(<AddTodoForm onSubmit={onSubmit} />);
  fireEvent.submit(getByTestId("form"));
  expect(onSubmit).toHaveBeenCalled();
});

test("input clears after adding new todo", () => {
  const onSubmit = jest.fn();
  const { getByTestId, getByRole } = render(
    <AddTodoForm onSubmit={onSubmit} />
  );
  const input = getByRole("textbox");
  fireEvent.change(input, { target: { value: "a" } });
  fireEvent.submit(getByTestId("form"));
  expect(input.value).toBe("");
});
