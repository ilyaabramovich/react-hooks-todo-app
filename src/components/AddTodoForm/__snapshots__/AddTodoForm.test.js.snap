// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<DocumentFragment>
  <form
    class="add-todo-form"
    data-testid="form"
  >
    <div
      class="add-todo-form__control"
    >
      <input
        class="input add-todo-form__input add-todo-form__input_fullwidth"
        placeholder="New thing to do"
        type="text"
        value=""
      />
    </div>
    <button
      class="button add-todo-form__button add-todo-form__button_fullwidth"
      disabled=""
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
